elkodon/src/compiletests.rs

```diff
@@ -1,21 +1,43 @@
 /// ```compile_fail
 /// use elkodon::prelude::*;
-/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
-/// # let service_name = ServiceName::new(b"My/Funk/ServiceName").unwrap();
-/// #
-/// # let service = zero_copy::Service::new(&service_name)
-/// #     .publish_subscribe()
-/// #     .open_or_create::<u64>()?;
-/// #
-/// # let publisher = service.publisher().create()?;
-///
-/// let mut sample = publisher.loan()?;
+/// fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// let service_name = ServiceName::new(b"My/Funk/ServiceName").unwrap();
+///
+/// let service = zero_copy::Service::new(&service_name)
+///     .publish_subscribe()
+///     .open_or_create::<u64>()?;
+///
+/// let publisher = service.publisher().create()?;
+///
+/// let mut sample = publisher.loan_uninit()?;
 /// sample.payload_mut().write(1234);
 ///
 /// publisher.send(sample)?; // should fail to compile
```

Comment:
Can you please explain why it failed and mention that the user has to call `assume_init` before `send`ing a sample in a comment before the line that leads to the failure.